pageserver/src/layered_repository.rs

```diff
@@ -2133,15 +2134,18 @@ impl LayeredTimeline {
             if let Some(pitr_cutoff_timestamp) = now.checked_sub(pitr) {
                 let pitr_timestamp = to_pg_timestamp(pitr_cutoff_timestamp);
 
-                match timeline.find_lsn_for_timestamp(pitr_timestamp)? {
-                    LsnForTimestamp::Present(lsn) => pitr_cutoff_lsn = lsn,
-                    LsnForTimestamp::Future(lsn) => {
+                match timeline.find_lsn_for_timestamp(pitr_timestamp) {
+                    Ok(LsnForTimestamp::Present(lsn)) => pitr_cutoff_lsn = lsn,
+                    Ok(LsnForTimestamp::Future(lsn)) => {
                         debug!("future({})", lsn);
                         pitr_cutoff_lsn = cutoff;
                     }
-                    LsnForTimestamp::Past(lsn) => {
+                    Ok(LsnForTimestamp::Past(lsn)) => {
                         debug!("past({})", lsn);
                     }
+                    Err(_) => {
+                        debug!("no commits found");
+                    }
```

Comment:
Why this change?

Reply:
It cause failures in test_branch_and_async_gc at the beginning: find_lsn_for_timestamp returns error if there are no transactions and it can not locate timestamp.

Reply:
After this discussion here, commit f09c09438a307f88eeb6bbaf85069022623e35d6 added a `LsnForTimestamp::Nodata` case for this, so I believe we don't need to do anything about it in this PR anymore.